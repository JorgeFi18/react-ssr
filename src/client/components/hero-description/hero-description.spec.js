import React from 'react';
import renderer from 'react-test-renderer';
import HeroDescription from './hero-description.component';

describe('Hero Description component', () => {

    it('Trigger should be called', ()=> {
        const handleClick = jest.fn();

        const component = renderer.create(
            <HeroDescription onClose={handleClick} />
        )

        const tree = component.toJSON();
        const [ button ] = tree.children;
        button.props.onClick();

        expect(handleClick).toHaveBeenCalled();
    });

    it('Description should match with snapshot', () => {
        const component = renderer.create(
            <HeroDescription description='Cool description' />
        )

        const tree = component.toJSON();
        const [ , description ] = tree.children;

        expect(description).toMatchSnapshot();
    });
});