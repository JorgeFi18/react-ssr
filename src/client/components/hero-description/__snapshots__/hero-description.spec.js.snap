// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Description component Description should match with snapshot 1`] = `
<p
  className="description__text"
>
  Cool description
</p>
`;
